lib/context.js

```diff
@@ -686,10 +690,12 @@ api.getInitialContext = options => {
     // create term selections for each mapping in the context, ordered by
     // shortest and then lexicographically least
     const mappings = activeCtx.mappings;
-    const terms = Object.keys(mappings).sort(_compareShortestLeast);
-    for(let i = 0; i < terms.length; ++i) {
-      const term = terms[i];
-      const mapping = mappings[term];
+    const terms = [...mappings.keys()].sort(_compareShortestLeast);
+    for(const term of terms) {
```

Comment:
would it be more performance to just do `get` first and then check for `undefined` or `null` or whatever?

Reply:
I haven't tested the difference, but it probably is faster when that pattern is ok.  I didn't consider it enough.  I don't think `undefined` is a value so perhaps that could be used.

Might also be faster to sort `mapping.entries()` and iterate through `[key, mapping]` pairs to just avoid `has` and `get` calls.